wgpu-hal/src/vulkan/adapter.rs

```diff
@@ -788,17 +788,25 @@ impl super::InstanceShared {
             let mut capabilities = PhysicalDeviceCapabilities::default();
             capabilities.supported_extensions =
                 unsafe { self.raw.enumerate_device_extension_properties(phd).unwrap() };
+
+            // Set the effective api version before we test for extension support
+            capabilities.effective_api_version = self
+                .driver_api_version
+                .min(capabilities.properties.api_version);
```

Comment:
Moving this up here won't work as `capabilities.properties.api_version` will be 0.

I opened https://github.com/gfx-rs/wgpu/pull/4252 as I also wanted to decouple the instance and device API versions for some time now.